libnopegl/src/pgcraft.c

```diff
@@ -1400,12 +1400,12 @@ struct pgcraft *ngli_pgcraft_create(struct ngl_ctx *ctx)
 
     ngli_darray_init(&s->pipeline_info.desc.textures,   sizeof(struct pipeline_texture_desc),   0);
     ngli_darray_init(&s->pipeline_info.desc.buffers,    sizeof(struct pipeline_buffer_desc),    0);
-    ngli_darray_init(&s->pipeline_info.desc.attributes, sizeof(struct pipeline_attribute_desc), 0);
+    ngli_darray_init(&s->pipeline_info.desc.attributes, sizeof(struct pgcraft_attribute_info), 0);
     ngli_darray_init(&s->pipeline_info.desc.vertex_buffers, sizeof(struct vertex_buffer_layout), 0);
```

Comment:
Mmh, so now in `info.desc` we have `_desc`, `_info` and `_layout`. What's the short to medium term naming consistency we're targeting here?

Reply:
First of all, I renamed the structure with the suffix _info to make it consistent with the pgcraft API (pgcraft_texture_info, pgcraft_compat_info).
In the upcoming change I plan to make:
- pipeline_texture_desc merged into pgcraft_texture_info
- pipeline_buffer_desc - >pgcraft_buffer_info
I will introduce the bindgroup API with the following new structs:
- struct bindgroup_layout { bindgroup_layout_entry[] }
- struct bindgroup {}

So at the end, info.desc will contains:
- pgcraft_texture_info
- pgcraft_buffer_info
- pgcraft_attribute_info
- bindgroup_layout_entry[]
- vertex_bufferlayout[]

This *may* be even more simplified using hmaps as most of the _info structure are still there to associate the name with a target (bindgroup_layout_entry or vertex_buffer_layout).

On another topic:
- rendertarget_desc -> rendertarget_layout
- graphicstate -> graphics_state

If you think this renaming introduce too much confusion for the time being, it can happen later on. I can use vertex_buffer_desc instead of vertex_buffer_layout and pgcraft_attribute_desc.

Note that the new naming convention I introduced is directly from wgpu.

Reply:
OK thanks, looking forward to it